// aiiot-center-npm-pkg/dist/components/Graph.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
var _recharts = require("recharts");
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _templateObject;
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _taggedTemplateLiteral(e, t) { return t || (t = e.slice(0)), Object.freeze(Object.defineProperties(e, { raw: { value: Object.freeze(t) } })); }
var ChartContainer = _styledComponents["default"].div(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  width: ", ";\n  height: ", ";\n  background-color: ", ";\n  padding: 20px;\n  border-radius: 8px;\n  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);\n"])), function (_ref) {
  var width = _ref.width;
  return width ? "".concat(width, "px") : "100%";
}, function (_ref2) {
  var height = _ref2.height;
  return height ? "".concat(height, "px") : "300px";
}, function (_ref3) {
  var bg = _ref3.bg;
  return bg || "#ffffff";
});
var Graph = function Graph(_ref4) {
  var data = _ref4.data,
    width = _ref4.width,
    height = _ref4.height,
    bg = _ref4.bg;
  return /*#__PURE__*/_react["default"].createElement(ChartContainer, {
    width: width,
    height: height,
    bg: bg
  }, /*#__PURE__*/_react["default"].createElement(_recharts.ResponsiveContainer, {
    width: "100%",
    height: "100%"
  }, /*#__PURE__*/_react["default"].createElement(_recharts.LineChart, {
    data: data
  }, /*#__PURE__*/_react["default"].createElement(_recharts.CartesianGrid, {
    strokeDasharray: "3 3"
  }), /*#__PURE__*/_react["default"].createElement(_recharts.XAxis, {
    dataKey: "name"
  }), /*#__PURE__*/_react["default"].createElement(_recharts.YAxis, null), /*#__PURE__*/_react["default"].createElement(_recharts.Tooltip, null), /*#__PURE__*/_react["default"].createElement(_recharts.Line, {
    type: "monotone",
    dataKey: "value",
    stroke: "#8884d8",
    strokeWidth: 2
  }))));
};
var _default = exports["default"] = Graph;
