// aiiot-center-npm-pkg/dist/components/SectionComponents.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Section = exports.Image = exports.Figure = exports.Div = exports.Card = void 0;
var _react = _interopRequireDefault(require("react"));
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _ThemeProvider = require("../theme/ThemeProvider");
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _taggedTemplateLiteral(e, t) { return t || (t = e.slice(0)), Object.freeze(Object.defineProperties(e, { raw: { value: Object.freeze(t) } })); }
var StyledDiv = _styledComponents["default"].div(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  padding: ", ";\n  margin: ", ";\n  display: ", ";\n  background-color: ", ";\n  color: ", ";\n"])), function (_ref) {
  var padding = _ref.padding;
  return padding || "16px";
}, function (_ref2) {
  var margin = _ref2.margin;
  return margin || "0";
}, function (_ref3) {
  var display = _ref3.display;
  return display || "block";
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.background;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.text;
});
var Div = exports.Div = function Div(props) {
  var _useTheme = (0, _ThemeProvider.useTheme)(),
    theme = _useTheme.theme;
  return /*#__PURE__*/_react["default"].createElement(StyledDiv, _extends({
    theme: theme
  }, props));
};
var StyledSection = _styledComponents["default"].section(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n  width: 100%;\n  padding: ", ";\n  background-color: ", ";\n"])), function (_ref6) {
  var padding = _ref6.padding;
  return padding || "20px";
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.sectionBg;
});
var Section = exports.Section = function Section(props) {
  var _useTheme2 = (0, _ThemeProvider.useTheme)(),
    theme = _useTheme2.theme;
  return /*#__PURE__*/_react["default"].createElement(StyledSection, _extends({
    theme: theme
  }, props));
};
var StyledCard = _styledComponents["default"].div(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n  padding: 20px;\n  border-radius: 8px;\n  box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1);\n  background-color: ", ";\n"])), function (_ref8) {
  var theme = _ref8.theme;
  return theme.cardBg;
});
var Card = exports.Card = function Card(props) {
  var _useTheme3 = (0, _ThemeProvider.useTheme)(),
    theme = _useTheme3.theme;
  return /*#__PURE__*/_react["default"].createElement(StyledCard, _extends({
    theme: theme
  }, props));
};
var StyledFigure = _styledComponents["default"].figure(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n  margin: 0;\n  padding: 0;\n  display: flex;\n  flex-direction: column;\n  align-items: center;\n"])));
var Figure = exports.Figure = function Figure(props) {
  return /*#__PURE__*/_react["default"].createElement(StyledFigure, props);
};
var StyledImage = _styledComponents["default"].img(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n  max-width: 100%;\n  height: auto;\n  border-radius: ", ";\n"])), function (_ref9) {
  var borderRadius = _ref9.borderRadius;
  return borderRadius || "4px";
});
var Image = exports.Image = function Image(props) {
  return /*#__PURE__*/_react["default"].createElement(StyledImage, props);
};
