// aiiot-center-npm-pkg/dist/components/Typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Typography = exports.Title = exports.Text = exports.Bold = void 0;
var _react = _interopRequireDefault(require("react"));
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _templateObject, _templateObject2, _templateObject3, _templateObject4;
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _taggedTemplateLiteral(e, t) { return t || (t = e.slice(0)), Object.freeze(Object.defineProperties(e, { raw: { value: Object.freeze(t) } })); }
var StyledTypography = _styledComponents["default"].p(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  font-size: ", ";\n  color: ", ";\n  line-height: 1.5;\n"])), function (_ref) {
  var size = _ref.size;
  return size ? "".concat(size, "px") : "16px";
}, function (_ref2) {
  var color = _ref2.color;
  return color || "#333333";
});
var StyledText = _styledComponents["default"].span(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n  font-size: ", ";\n  color: ", ";\n"])), function (_ref3) {
  var size = _ref3.size;
  return size ? "".concat(size, "px") : "14px";
}, function (_ref4) {
  var color = _ref4.color;
  return color || "#333333";
});
var StyledBold = _styledComponents["default"].strong(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n  font-size: ", ";\n  color: ", ";\n  font-weight: bold;\n"])), function (_ref5) {
  var size = _ref5.size;
  return size ? "".concat(size, "px") : "16px";
}, function (_ref6) {
  var color = _ref6.color;
  return color || "#000000";
});
var StyledTitle = _styledComponents["default"].h2(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n  font-size: ", ";\n  font-weight: bold;\n  color: ", ";\n"])), function (_ref7) {
  var size = _ref7.size;
  return size ? "".concat(size, "px") : "28px";
}, function (_ref8) {
  var color = _ref8.color;
  return color || "#222222";
});
var Typography = exports.Typography = function Typography(_ref9) {
  var children = _ref9.children,
    size = _ref9.size,
    color = _ref9.color;
  return /*#__PURE__*/_react["default"].createElement(StyledTypography, {
    size: size,
    color: color
  }, children);
};
var Text = exports.Text = function Text(_ref10) {
  var children = _ref10.children,
    size = _ref10.size,
    color = _ref10.color;
  return /*#__PURE__*/_react["default"].createElement(StyledText, {
    size: size,
    color: color
  }, children);
};
var Bold = exports.Bold = function Bold(_ref11) {
  var children = _ref11.children,
    size = _ref11.size,
    color = _ref11.color;
  return /*#__PURE__*/_react["default"].createElement(StyledBold, {
    size: size,
    color: color
  }, children);
};
var Title = exports.Title = function Title(_ref12) {
  var children = _ref12.children,
    size = _ref12.size,
    color = _ref12.color;
  return /*#__PURE__*/_react["default"].createElement(StyledTitle, {
    size: size,
    color: color
  }, children);
};
